File: app/src/main/kotlin/com/kotlinextension/data/db/DatabaseSource.java
package com.kotlinextension.data.db;

import java.util.List;

import io.reactivex.Flowable;

public interface DatabaseSource<T> {
    Flowable<T> get(T t);
    Flowable<List<T>> getAll();
    void insert(T t);
    void delete(T t);
    void update(T t);
}

File: app/src/main/kotlin/com/kotlinextension/data/db/DatabaseSource.kt
package com.kotlinextension.data.db

import io.reactivex.Flowable

interface DatabaseSource<T> {
    fun getAll() :Flowable<MutableList<T>>
    fun get(bean :T) :Flowable<T>
    fun insert(bean :T)
    fun delete(bean :T)
    fun update(bean :T)
}
